test/pylib/rest_client.py

```diff
@@ -0,0 +1,163 @@
+        resp = await self.session.get(self._resource_uri(resource, host))
+        if resp.status != 200:
+            text = await resp.text()
+            raise Exception(f"status code: {resp.status}, body text: {text}")
+        return resp
+
+    async def get_text(self, resource: str, host: str) -> str:
+        """Fetch remote resource text response or raise"""
+        resp = await self.get(resource, host)
+        return await resp.text()
+
+    async def put_json(self, resource: str, json: dict, host: str) \
+            -> aiohttp.ClientResponse:
+        """Put JSON"""
+        return await self.session.request(method="PUT", url=self._resource_uri(resource, host),
+                                          json=json)
+
+    async def post(self, resource: str, params: Dict[str, str], host: str) \
+            -> aiohttp.ClientResponse:
+        """Post to remote resource or raise"""
+        resp = await self.session.post(self._resource_uri(resource, host), params=params)
+        if resp.status != 200:
+            text = await resp.text()
+            raise Exception(f"status code: {resp.status}, body text: {text}")
+        return resp
+
+
+class ScyllaRESTAPIClient():
+    """Async Scylla REST API client"""
+
+    def __init__(self, port: int = 10000):
+        self.cli = TCPRESTClient(port)
+
+    async def stop(self):
+        """Close session"""
+        await self.cli.stop()
+
+    async def get_server_uuid(self, server_id: str):
+        """Get server id (UUID)"""
+        host_uuid = await self.cli.get_text("/storage_service/hostid/local", host=server_id)
+        host_uuid = host_uuid.lstrip('"').rstrip('"')
+        return host_uuid
+
+    # TODO: change to UUID
+    async def wait_for_host_known(self, dst_server_id: str, expect_server_id: str,
+                                  timeout: float = 30.0, sleep: float = 0.05) -> bool:
+        """Checks until dst_server_id knows about expect_server_id, with timeout"""
+        max_time = time() + timeout
+        while True:
+            resp_body = await self.cli.get_text("/storage_service/host_id", dst_server_id)
+            if expect_server_id in resp_body:
+                logger.info("wait_for_host_known [{%s}] found, SUCCESS", expect_server_id)
+                return True
+            if time() > max_time:
+                logger.info("wait_for_host_known [{%s}] timeout", expect_server_id)
+                return False
+            logger.debug("wait_for_host_known [{%s}] not found, sleeping {sleep}", expect_server_id)
+            await asyncio.sleep(sleep)
+
+    async def remove_node(self, initiator_ip: str, server_uuid: str) -> bool:
+        """Initiate remove node of server_uuid in initiator initiator_ip"""
+        resp = await self.cli.post("/storage_service/remove_node", params={"host_id": server_uuid},
+                                   host=initiator_ip)
+        logger.debug("remove_node status %s for %s", resp.status, server_uuid)
```

Comment:
`TCPRESTClient.post` already checks for 200 status code, so `resp.status == 200` is redundant. Seems that we can leave void as return type for `remove_node` and `wait_for_remove`, I'm ok with getting exception if something went wrong.

Regarding http server handling, maybe we can have general logic, which will return 500/log everything in case of exceptions?

Reply:
done in https://github.com/scylladb/scylladb/commit/c6d671384519aebc94439e2c64428e3e69d53441